src/main/java/mekanism/client/ClientTickHandler.java

```diff
@@ -241,6 +248,19 @@ public void tickStart() {
                 }
             }
 
+            if (isHydroEnhancementOn(minecraft.player)) {
```

Comment:
I haven't tested yet but part of me feels like this might not work quite right especially on dedicated servers as the effect is only being added client side. (Honestly I am slightly unsure why Aidan made it so the night vision potion is only client side instead of being fully server side controlled unless it is purely given the fact it doesn't really matter for that one given if a client modifies it to adjust how things behave there is a lot of the night vision logic that is client side anyway due to adjusting fog and color rendering)

Reply:
So this is actually very similar in that all movement is controlled by the client. The server will kick if it thinks the movement is "flying" but as this just affects swim speed I haven't had it have issues yet, and I've tested on a server.